packages/@stylexjs/babel-plugin/__tests__/transform-stylex-props-test.js

```diff
@@ -231,6 +231,451 @@ describe('@stylexjs/babel-plugin', () => {
+        const output = transform(`
+          import stylex from 'stylex';
+          import * as css from '@stylexjs/inline-css';
+          const styles = stylex.create({
+            base: { color: 'red', backgroundColor: 'white' },
+          });
+          stylex.props(styles.base, css.color.blue, css.backgroundColor.white);
+        `);
+        expect(output).toMatchInlineSnapshot(`
+          "import _inject from "@stylexjs/stylex/lib/stylex-inject";
+          var _inject2 = _inject;
+          import stylex from 'stylex';
+          import * as css from '@stylexjs/inline-css';
+          _inject2({
+            ltr: ".x1e2nbdu{color:red}",
+            priority: 3000
+          });
+          _inject2({
+            ltr: ".x12peec7{background-color:white}",
+            priority: 3000
+          });
+          _inject2({
+            ltr: ".xju2f9n{color:blue}",
+            priority: 3000
+          });
+          _inject2({
+            ltr: ".x12peec7{background-color:white}",
+            priority: 3000
+          });
+          ({
+            className: "xju2f9n x12peec7"
+          });"
+        `);
+      });
+
+      test('dynamic style', () => {
+        const inline = transform(`
+          import stylex from 'stylex';
+          import * as css from '@stylexjs/inline-css';
+          stylex.props(css.color(color));
+        `);
+        const local = transform(`
+          import stylex from 'stylex';
+          const styles = stylex.create({
+            color: (c) => ({ color: c }),
+          });
+          stylex.props(styles.color(color));
+        `);
+        expect(local).toMatchInlineSnapshot(`
+          "import _inject from "@stylexjs/stylex/lib/stylex-inject";
+          var _inject2 = _inject;
+          import stylex from 'stylex';
+          _inject2({
+            ltr: ".x14rh7hd{color:var(--x-color)}",
+            priority: 3000
+          });
+          _inject2({
+            ltr: "@property --x-color { syntax: \\"*\\"; inherits: false;}",
+            priority: 0
+          });
+          const styles = {
+            color: c => [{
+              kMwMTN: c != null ? "x14rh7hd" : c,
+              $$css: true
+            }, {
+              "--x-color": c != null ? c : undefined
+            }]
+          };
+          stylex.props(styles.color(color));"
+        `);
+        expect(inline).toMatchInlineSnapshot(`
+          "import _inject from "@stylexjs/stylex/lib/stylex-inject";
+          var _inject2 = _inject;
+          import stylex from 'stylex';
+          import * as css from '@stylexjs/inline-css';
+          _inject2({
+            ltr: ".x14rh7hd{color:var(--x-color)}",
+            priority: 3000
+          });
+          _inject2({
+            ltr: "@property --x-color { syntax: \\"*\\"; inherits: false;}",
+            priority: 0
+          });
+          stylex.props([{
+            "color": color != null ? "x14rh7hd" : color,
+            "$$css": true
+          }, {
```

Comment:
It is important that that the style objects created by the inline-css package are hoisted to the module level as a constant and not inlined like this. We already have a utility for doing this somewhere.